lib/bolt/node/ssh.rb

```diff
@@ -290,18 +290,32 @@ def _run_task(task, input_method, arguments)
           "PT_#{env}='#{val}'"
         end.join(' ')
       end
+     
+      use_sudo = @sudo || @run_as
 
-      with_remote_task(task, stdin) do |remote_path|
-        command = if export_args.empty?
-                    "'#{remote_path}'"
-                  else
-                    "#{export_args} '#{remote_path}'"
-                  end
-        output = execute(command, sudoable: true)
-        Bolt::TaskResult.from_output(output)
+      if use_sudo
+        with_remote_task(task, stdin) do |remote_path|
+          command = if export_args.empty?
```

Comment:
If we start command outside of this block this might be a bit cleaner

```
command = export_args.empty? ? '' :  "#{export_args} "
if use_sudo
  <snip>
  command += remote_path
else
  <snip>
   command += remote_path
  <snip>
```

Reply:
I don't think this is super important but if you're unhappy with the duplication this minimizes it to one line.